src/child.cpp

```diff
@@ -0,0 +1,470 @@
+                std::cout << lineo << "QSCAND: set: needs parameter!\n";
+#ifdef _WIN32
+                send(arg->connfd, lineo, strlen(lineo), 0);
+#else
+                write(arg->connfd, lineo, strlen(lineo));
+#endif
+                continue;
+            }
+            if (!strncmp(linet, "dev=", 4)) {
+                linet+=4;
+                len = strlen(linet);
+                if (len<sizeof(device)) {
+                    strncpy(device, linet, len-1);
+                } else {
+                    std::cout << lineo << "QSCAND: too long device name!\n";
+#ifdef _WIN32
+                    send(arg->connfd, lineo, strlen(lineo), 0);
+#else
+                    write(arg->connfd, lineo, strlen(lineo));
+#endif
+                }
+            } else if (!strncmp(linet, "test=", 5)) {
+                linet+=5;
+                len = strlen(linet);
+                if (len<sizeof(test)) {
+                    strncpy(test, linet, len-1);
+                } else {
+                    std::cout << lineo << "QSCAND: too long test name!\n";
+#ifdef _WIN32
+                    send(arg->connfd, lineo, strlen(lineo), 0);
+#else
+                    write(arg->connfd, lineo, strlen(lineo));
+#endif
+                }
+            } else if (!strncmp(linet, "speed=", 6)) {
+                linet+=6;
+                speed = atol(linet);
+            } else if (!strncmp(linet, "simul=", 6)) {
+                linet+=6;
+                WT_simul = atol(linet) ? 1 : 0;
+            } else {
+                std::cout << lineo << "QSCAND: set: invalid parameter!\n";
+#ifdef _WIN32
+                send(arg->connfd, lineo, strlen(lineo), 0);
+#else
+                write(arg->connfd, lineo, strlen(lineo));
+#endif
+            }
+            continue;
+        } else {
+        //  sprintf(lineo, "str len: %d, cmd len: %d. '%d'\n", n, strlen(linei), linei);
+            //sprintf(lineo, "str len: %d, cmd len: %d, last %02x %02x\n", n, strlen(linei), linei[n-2], linei[n-1]);
+        //  write(arg->connfd, lineo, strlen(lineo));
+            std::cout << lineo << "QSCAND: invalid command. try \"help\"\n";
+#ifdef _WIN32
+            send(arg->connfd, lineo, strlen(lineo), 0);
+#else
+            write(arg->connfd, lineo, strlen(lineo));
+#endif
+        }
+        
+        if (mode != none) {
+            int    argc = 0;
+            char **argv = (char**) malloc(sizeof(char*));
+            argv[0] = NULL;
+            if ( ((mode == scan) || (mode == dinfo) || (mode == minfo))  && !strlen(device)) {
+#ifdef DAEMON_EN
+                if (daemonized)
+                // TODO: Redirect cout statements to file to emulate syslog behavior
+                    std::cout << LOG_WARNING << "QSCAND: No device specified!\n";
+                else
+#endif
+                    std::cout << "QSCAND: No device specified!\n";
+                _exit(0);
+            }
+            if ( (mode == scan) && !strlen(test)) {
```

Comment:
Codacy found an issue: [Does not handle strings that are not \0-terminated; if given one it may perform an over-read (it could cause a crash if unprotected) (CWE-126).  ](https://app.codacy.com/manual/jeremiahwilson777/qpxtool/pullRequest?prid=6191247)